### PR TITLE
fix: Collapse hashes for dynamically generated javassist classes/file…

### PR DESCRIPTION
**Pull request by bretthoerner**
…names

#8165

### CODE REVIEW

File: src/sentry/interfaces/stacktrace.py [hunk 1]
```diff
@@ -183,15 +187,20 @@ def remove_filename_outliers(filename, platform=None):
     # contain things like /Users/foo/Dropbox/...
     if platform == 'cocoa':
         return posixpath.basename(filename)
+    elif platform == 'java':
+        filename = _java_assist_enhancer_re.sub(r'\1<auto>', filename)
```
Comment (dcramer) [inline src/sentry/interfaces/stacktrace.py]: you sure your test is actually hitting this path? codecov says not

Comment (bretthoerner) [reply src/sentry/interfaces/stacktrace.py]: Oh yeah, I think having a module set makes the module "win." Need another case.

Comment (bretthoerner) [reply src/sentry/interfaces/stacktrace.py]: Fixed.

**Comment by bretthoerner** [comment]
Bump.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**